src/Veldrid.Utilities/ObjParser.cs

```diff
@@ -12,58 +11,111 @@ namespace Veldrid.Utilities
     /// </summary>
     public class ObjParser
     {
-        private static readonly string[] s_newline = new string[] { "\n" };
-        private static readonly char[] s_whitespaceChars = new char[] { ' ' };
-        private static readonly char[] s_slashChar = new char[] { '/' };
+        private const int InitialReadBufferSize = 2048;
+
+        private static readonly char[] _whitespaceChar = new[] { ' ' };
+        private static readonly char _slashChar = '/';
 
         private readonly ParseContext _pc = new ParseContext();
+        private char[] _readBuffer;
 
         /// <summary>
         /// Parses an <see cref="ObjFile"/> from the given raw text lines.
         /// </summary>
         /// <param name="lines">The text lines of the OBJ file.</param>
         /// <returns>A new <see cref="ObjFile"/>.</returns>
-        public ObjFile Parse(string[] lines)
+        public ObjFile Parse(IEnumerable<string> lines)
```

Comment:
This is a breaking change to the method signature, and needs to be reverted to maintain backwards compatibility.